src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -269,7 +284,15 @@ private IHost BuildHost(bool skipHostStartup = false, bool skipHostJsonConfigura
 
         private ILogger GetHostLogger()
         {
-            var hostLoggerFactory = _host?.Services.GetService<ILoggerFactory>();
+            ILoggerFactory hostLoggerFactory = null;
+            try
+            {
+                hostLoggerFactory = _host?.Services.GetService<ILoggerFactory>();
```

Comment:
Running locally, some of the failures I saw were due to ContainerExceptions occurring due to access of a disposed host container.